backend/api/systemOauthClient.ts

```diff
@@ -1,31 +1,87 @@
 import querystring from 'querystring'
-import clientFactory from './oauthEnabledClient'
 
-import { oauthApiFactory } from './oauthApi'
-import config from '../config'
+import { createClient } from 'redis'
+import { promisify } from 'util'
 import logger from '../log'
 
-export const getClientCredentialsTokens = async (username) => {
+let getRedisAsync
+let setRedisAsync
+let oauthClient
+
+export const getTokenStore = (config) => {
+  const { enabled, host, port, password } = config.redis
+  if (!enabled || !host) return null
+
+  const client = createClient({
+    host,
+    port,
+    password,
+    tls: config.app.production ? {} : false,
+  })
+
+  client.on('error', (e: Error) => logger.error('Redis client error', e))
```

Comment:
Doesnt the e parameter come first?

Reply:
You mean in the `error` function parameters? Hmm.. yes.. which means the other 2 places in DPS are wrong as well. I'll fix them

Good spot!